aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -183,7 +183,14 @@ def _get_hashed_idempotency_key(self, lambda_event: Dict[str, Any]) -> str:
 
     @staticmethod
     def is_missing_idempotency_key(data) -> bool:
-        return data is None or not data or all(x is None for x in data)
+        if data is None:
+            return True
+        elif not data:
+            return True
+        elif type(data).__name__ in ("tuple", "list", "dict"):
+            return all(x is None for x in data)
+        else:
+            return False
```

Comment:
Perhaps no else and simply return False if flow above doesn't yield anything?

return False

I think we can do with two Ifs only